content/posts/lets-code-contact-form-in-phoenix-part-one.md

```diff
@@ -31,17 +31,27 @@ end
 
 Here, I define a `Support.Message` module to model the user's message. Since this is an internal module to the `Support` context, I specify `@moduledoc false`. [This convention](https://hexdocs.pm/elixir/writing-documentation.html#hiding-internal-modules-and-functions) hides the module from our application's documentation. I typically document modules that are part of the public interface of my application and hide the rest.
```

Comment:
I think that makes sense. I'll play around with the arrangement and see what I come up with.

Reply:
I went with:

> Here, I define a `Support.Message` module to model the user's, you guessed it, support message. The structure of this module should contain the email, subject, and body fields. The `Message` module is an internal module to be used by the `Support` context only. The `Support` module, which will be defined in a future post, is the public interface for and the boundary around all support-related functionality in the application.

> Stepping through the code, I start by specifying `@moduledoc false`. [This convention](https://hexdocs.pm/elixir/writing-documentation.html#hiding-internal-modules-and-functions) hides this internal module from our application's documentation. I typically document modules that are part of the public interface of the application and hide the rest.

> The `@type t` convention documents the type this module models. In this case, it's a struct of this module, denoted with the [`__MODULE__` convenience function](https://hexdocs.pm/elixir/Kernel.SpecialForms.html#__MODULE__/0), and containing `email`, `subject`, and `body` attributes. Simple enough so far. Note, this module will not compile because a struct has not been defined. Next, I will add a schema using [the Ecto library](https://hexdocs.pm/ecto/Ecto.html) which will give us the struct and allow this module to be compiled.

@tmr08c, what do you think?